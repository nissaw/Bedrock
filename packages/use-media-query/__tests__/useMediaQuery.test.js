import React from "react";
import ReactDOM from "react-dom/";
import { act } from "react-dom/test-utils";

import useMediaQuery from "../src";

let matches;
const HookWrapper = ({ width = 320 }) => {
  matches = useMediaQuery(`(max-width:${width}px)`);

  return null;
};

describe("useMediaQuery", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    matches = undefined;
    document.body.removeChild(container);
    container = null;
  });
  test("useMediaQuery is not null", () => {
    expect(useMediaQuery).toBeTruthy();
  });
  it("should call window.matchMedia", () => {
    act(() => {
      ReactDOM.render(<HookWrapper />, container);
    });

    expect(window.matchMedia).toBeCalled();
    expect(window.matchMedia.mock.calls).toMatchSnapshot();
  });

  it("should update matches when listener is called", () => {
    let listeners = [];
    window.matchMedia.mockImplementation((query) => ({
      matches: true,
      media: query,
      addListener: jest.fn((fn) => {
        listeners = listeners.concat(fn);
      }),
      removeListener: jest.fn((fn) => {
        listeners = listeners.filter((listener) => listener !== fn);
      }),
    }));

    act(() => {
      ReactDOM.render(<HookWrapper />, container);
    });

    listeners[0]({ matches: false });

    expect(matches).toBe(false);
    window.matchMedia.mockRestore();
  });

  it("wont update matches if unmounted", () => {
    let listeners = [];
    window.matchMedia.mockImplementation((query) => ({
      matches: true,
      media: query,
      addListener: jest.fn((fn) => {
        listeners = listeners.concat(fn);
      }),
      removeListener: jest.fn(),
    }));

    act(() => {
      ReactDOM.render(<HookWrapper />, container);
    });

    act(() => {
      ReactDOM.unmountComponentAtNode(container);
    });

    listeners[0]({ matches: false });

    expect(matches).toBe(true);
    window.matchMedia.mockRestore();
  });

  it("returns false if matchMedia not on window", () => {
    const matchMedia = window.matchMedia;
    window.matchMedia = undefined;
    act(() => {
      ReactDOM.render(<HookWrapper />, container);
    });

    expect(matches).toBe(false);
    window.matchMedia = matchMedia;
  });
});
