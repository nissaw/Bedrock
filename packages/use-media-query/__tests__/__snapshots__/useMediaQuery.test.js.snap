// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useMediaQuery should call window.matchMedia 1`] = `
Array [
  Array [
    "(max-width:320px)",
  ],
  Array [
    "(max-width:320px)",
  ],
]
`;
